File: Medium/Sort an array of 0s, 1s and 2s/sort-an-array-of-0s-1s-and-2s.java
//{ Driver Code Starts
//Initial template for Java

import java.io.*;
import java.util.*;


// } Driver Code Ends
//User function template for Java

class Solution
{
    public static void sort012(int a[], int n)
    {
        int zero = 0, one = 0, two = 0;
        for(int i : a) {
            if(i == 0)
                zero++;
            else if(i == 1)
                one++;
            else if(i == 2)
                two++;
        }
        
        
        for(int i = 0; i < a.length; i++) {
            if(zero-- > 0)
                a[i] = 0;
            else if(one-- > 0)
                a[i] = 1;
            else if(two-- > 0)
                a[i] = 2;
        }
    }
}

//{ Driver Code Starts.

class GFG {
    
    public static void main (String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim()); //Inputting the testcases
        while(t-->0){
            int n = Integer.parseInt(br.readLine().trim());
            int arr[] = new int[n];
            String inputLine[] = br.readLine().trim().split(" ");
            for(int i=0; i<n; i++){
                arr[i] = Integer.parseInt(inputLine[i]);
            }
            Solution ob=new Solution();
            ob.sort012(arr, n);
            StringBuffer str = new StringBuffer();
            for(int i=0; i<n; i++){
                str.append(arr[i]+" ");
            }
            System.out.println(str);
        }
    }
}


// } Driver Code Ends

File: Medium/Sort an array of 0s, 1s and 2s/sort-an-array-of-0s-1s-and-2s.py
#User function Template for python3

class Solution:
    def sort012(self,arr,n):
        zero = one = two = 0
        for i in arr:
            if i == 0:
                zero += 1
            elif i == 1:
                one += 1
            elif i == 2:
                two += 2
        for i in range(len(arr)):
            if zero > 0:
                arr[i] = 0
                zero -= 1
            elif one > 0:
                arr[i] = 1
                one -= 1
            elif two > 0:
                arr[i] = 2
                two -= 1
        return arr



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t=int(input())
    for _ in range(t):
        n=int(input())
        arr=[int(x) for x in input().strip().split()]
        ob=Solution()
        ob.sort012(arr,n)
        for i in arr:
            print(i, end=' ')
        print()

# } Driver Code Ends